src/plugins/intel_gna/src/transformations/handle_transposes_around_matmul.cpp

```diff
@@ -150,11 +150,13 @@ HandleTransposeBeforeMatMul::HandleTransposeBeforeMatMul() {
             return false;
         }
 
-        auto matmul_node = matmul_iter->second.get_node_shared_ptr();
+        auto matmul_node = std::dynamic_pointer_cast<ngraph::opset8::MatMul>(matmul_iter->second.get_node_shared_ptr());
+        if (!matmul_node)
+            return false;
         auto transpose_reshape_it = pattern_map.find(transpose1);
         if (transpose_reshape_it != std::end(pattern_map)) {
             ReplaceTransposeWithReshape(transpose_reshape_it->second.get_node_shared_ptr());
-        } else {
+        } else if (!matmul_node->get_transpose_a()) {
```

Comment:
Could you please explain why is this logic change required?

Reply:
I disabled the tests for now.

Reply:
ticket number: CVS-123306